src/scenes/CommunityPage/CommunityPageView.tsx

```diff
@@ -12,33 +12,73 @@ import TextButton from 'components/core/Button/TextButton';
 import breakpoints from 'components/core/breakpoints';
 import TokenHolderList from 'components/TokenHolderList/TokenHolderList';
 import formatUrl from 'utils/formatUrl';
-import { VStack } from 'components/core/Spacer/Stack';
+import { HStack, VStack } from 'components/core/Spacer/Stack';
+import colors from 'components/core/colors';
+import LayoutToggleButton from './LayoutToggleButton';
+import { DisplayLayout } from 'components/core/enums';
+import CommunityHolderGrid from 'components/CommunityHolderGrid/CommunityHolderGrid';
+import isFeatureEnabled, { FeatureFlag } from 'utils/graphql/isFeatureEnabled';
+import { CommunityPageViewQueryFragment$key } from '__generated__/CommunityPageViewQueryFragment.graphql';
 
 type Props = {
   communityRef: CommunityPageViewFragment$key;
+  queryRef: CommunityPageViewQueryFragment$key;
 };
 
-export default function CommunityPageView({ communityRef }: Props) {
+export default function CommunityPageView({ communityRef, queryRef }: Props) {
   const community = useFragment(
     graphql`
       fragment CommunityPageViewFragment on Community {
         name
         description
+        contractAddress {
+          address
+        }
         owners(first: 10000) @connection(key: "CommunityPageView_owners") {
           edges {
             node {
               ...TokenHolderListFragment
             }
           }
         }
+
+        tokensInCommunity(first: 10000) @connection(key: "CommunityPageView_tokensInCommunity") {
```

Comment:
Is our server going to be okay lol? 😅

Reply:
Might not 🧨. Pagination will saved this problem 😬

Reply:
hahah we'll definitely want pagination! it might be a bit different for grid vs. people views so i'll write up some notes on it